### PR TITLE
Reduce code duplication in macros

### PR DESCRIPTION
**Pull request by qkoziol**
Reduce copies of the macros and create generic versions for all lists

### CODE REVIEW

File: src/H5Cpkg.h [hunk 1]
```diff
@@ -82,424 +75,167 @@
...
+    ((entry_ptr)->list_next == NULL && (tail_ptr) != (entry_ptr)) ||        \
+    ((len) == 1 &&                                                           \
+     !((head_ptr) == (entry_ptr) && (tail_ptr) == (entry_ptr) &&             \
+       (entry_ptr)->list_next == NULL && (entry_ptr)->list_prev == NULL && \
+       (list_size) == (entry_ptr)->size                                      \
+      )                                                                      \
+    )                                                                        \
    ) {                                                                       \
-    HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, (fail_val), "DLL pre remove SC failed")     \
+    HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, (fail_val), "DLL pre insert SC failed")               \
```
Comment (derobins) [inline src/H5Cpkg.h]: The macro says REMOVE but the error message says "insert"

**Review by derobins — APPROVED** [review]
Need to change that error string, though

File: src/H5Cpkg.h (hunk 1 above)
Comment (qkoziol) [reply src/H5Cpkg.h]: Done

**Review by hyoklee — APPROVED** [review]

File: src/H5Cpkg.h [hunk 2]
```diff
@@ -82,424 +75,167 @@
...
-       )                                                                 \
-     ) ||                                                                \
-     ( ( (len) >= 1 ) &&                                                 \
-       ( ( (head_ptr) == NULL ) || ( (head_ptr)->prev != NULL ) ||       \
-         ( (tail_ptr) == NULL ) || ( (tail_ptr)->next != NULL )          \
-       )                                                                 \
-     )                                                                   \
-   ) {                                                                   \
-    HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, (fail_val), "DLL sanity check failed")  \
+#define H5C__GEN_DLL_SC(list_next, list_prev, head_ptr, tail_ptr, len, list_size, fail_val) \
```
Comment (fortnern) [inline src/H5Cpkg.h]: This macro doesn't seem to be used for anything

Comment (qkoziol) [reply src/H5Cpkg.h]: Ah, interesting...    I looked in older versions of the file and it doesn't appear used in the ones I've looked at either.   I'll remove it.

File: src/H5Cpkg.h [hunk 3]
```diff
@@ -82,424 +75,151 @@
...
-     ( ( (len) == 1 ) &&                                                     \
-       ( ! ( ( (head_ptr) == (entry_ptr) ) &&                                \
-             ( (tail_ptr) == (entry_ptr) ) &&                                \
-             ( (entry_ptr)->next == NULL ) &&                                \
-             ( (entry_ptr)->prev == NULL ) &&                                \
-             ( (list_size) == (entry_ptr)->size )                                 \
-           )                                                                 \
-       )                                                                     \
-     )                                                                       \
+#define H5C__GEN_DLL_PRE_REMOVE_SC(entry_ptr, list_next, list_prev, head_ptr, tail_ptr, len, list_size, fail_val) \
```
Comment (jhendersonHDF) [inline src/H5Cpkg.h:78]: It doesn't look like the new list_next and list_prev params are used in this macro. Were they supposed to replace entry_ptr->list_next/list_prev?

Comment (qkoziol) [reply src/H5Cpkg.h:78]: :-?

Comment (qkoziol) [reply src/H5Cpkg.h:78]: They look correct to me

Comment (qkoziol) [reply src/H5Cpkg.h:78]: I think I understand what you are saying - the way they are used: (entry_ptr)->list_next/prev is the required syntax to get the substitution correct by the compiler.

Comment (jhendersonHDF) [reply src/H5Cpkg.h:78]: Ah ok I see what you did now. Seeing only entry_ptr in parenthesis was a little misleading to me. Didn't realize the field accessed through entry_ptr is also genericized through the list_next and list_prev params.

Comment (qkoziol) [reply src/H5Cpkg.h:78]: Yeah, I wanted the put list_next/prev in parens also, but the compiler barfs on that

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**